File: LinguagemC/vetor_sena.c
/*
Algoritmos Resolvidos em C
Prof.José Augusto Cintra - http://josecintra.com/blog

Enunciado: Dados um palpite da sena e o resultado. Verificar quantos acertos o palpite obteve
*/
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
  int aposta[6] = {1,9,3,8,5,6};
  int result[6] = {1,2,3,4,5,6};
  int acertos = 0;
  int i_aposta,i_result;

  for(i_aposta = 0; i_aposta < 6; i_aposta++)
    for(i_result = 0; i_result < 6; i_result++){
      if (aposta[i_aposta] == result[i_result]) {
        acertos++;
      }
  }
  printf("Numero de acertos = %d",acertos);
  return 0;
}

File: Java/Vetor_Sena.java
/*
Algoritmos Resolvidos em Java
Prof.José Augusto Cintra - http://josecintra.com/blog

Enunciado: Dado um resultado da Sena, Leia um palpite e informe quantos acertos esse palpite obteve
*/

import java.util.*;

public class Vetor_Sena {
   public static void main(String[] args) {
      Scanner teclado = new Scanner(System.in);
      int[] aposta = new int[6];
      int[] result = { 1, 22, 34, 41, 55, 59 };
      int acertos = 0;
      // leitura da aposta
      for (int conta = 0; conta < aposta.length; conta++) {
         System.out.print("Informe número para aposta: ");
         aposta[conta] = teclado.nextInt();
      }
      // Verificar acertos
      for (int conta1 = 0; conta1 < result.length; conta1++)
         for (int conta2 = 0; conta2 < result.length; conta2++)
            if (aposta[conta1] == result[conta2])
               acertos++;
      // Imprimir resultado
      System.out.print("Número de acertos: " + acertos);
   }
}
